pelican_bib/bib.py

```diff
@@ -160,6 +201,159 @@ def add_publications(generator):
     generator.context['publications_lists'] = publications_lists
 
 
+class Bibliography(Directive):
+    """
+    Directive to embed bibliographies into articles/posts.
+
+    Usage::
+
+        .. bibliography:: PUBLICATIONS_SRC
+          :template: TEMPLATE_NAME (optional)
+          :options: TEMPLATE_OPTIONS_DICT (optional)
+          :class: CLASS_NAME (optional)
+          :filter_tag: FILTER_TAG (optional)
+          :pybtex_style_args: PYBTEX_STYLE_ARGS (optional)
+          :sorting_style: PYBTEX_SORTING_STYLE (optional)
+          :abbreviate_names: PYBTEX_ABBREVIATE_NAMES (optional)
+          :name_style: PYBTEX_NAME_STYLE (optional)
+
+    e.g.::
+
+        .. bibliography:: osm.bib
+          :template: publications_unsrt
+          :options: {'groupby_value': year}
+          :class: my-bib
+          :pybtex_style_args: {'sorting_style': 'author_year_title',
+                               'abbreviate_names': False,
+                               'name_style': 'lastfirst' }
+          :sorting_style: author_year_title
+          :abbreviate_names: False
+          :name_style: lastfirst
+    """
+    required_arguments = 0
+    optional_arguments = 100
```

Comment:
We are actually allowing an infinite number of files, as far as I got it, it is not too big of an issue to put a higher number here maybe?

```suggestion
     required_arguments, optional_arguments = 0,512
```

Reply:
Originally, the number of input files exactly one. Allowing more than one input file was a side-effect of the feature "BibTeX sources as content". 100 sounded more than enough for me. But we can set it to any arbitrary number larger than 1. :-)